rdk/rdk.py

```diff
@@ -3543,6 +3605,89 @@ def __tag_config_rule(self, rule_name, cfn_tags, my_session):
         )
         return response
 
+    def __get_lambda_layers(self,session,args,params):
+        layers = []
+        if 'SourceRuntime' in params:
+            if params['SourceRuntime'] in ['python3.6-lib', 'python3.7-lib', 'python3.8-lib']:
+                if args.generated_lambda_layer:
+                    lambda_layer_version = self.__get_existing_lambda_layer(session)
+                    if not lambda_layer_version:
+                        print(f"{session.region_name} --generated-lambda-layer flag received, but rdklib-layer not found in {session.region_name}. Creating one now")
+                        self.__create_new_lambda_layer(session)
+                        lambda_layer_version = self.__get_existing_lambda_layer(session)
+                    layers.append(lambda_layer_version)
+                elif args.rdklib_layer_arn:
+                    layers.append(args.rdklib_layer_arn)
+                else:
+                    rdk_lib_version = RDKLIB_LAYER_VERSION[session.region_name]
+                    rdklib_arn = RDKLIB_ARN_STRING.format(region=session.region_name, version=rdk_lib_version)
+                    layers.append(rdklib_arn)
+        return layers
+
+    def __get_existing_lambda_layer(self, session):
+        region = session.region_name
+        lambda_client = session.client("lambda")
+        print(f"[{region}]: Checking for Existing RDK Layer")
+        response = lambda_client.list_layer_versions(LayerName="rdklib-layer")
+        if response["LayerVersions"]:
+            return response['LayerVersions'][0]['LayerVersionArn']
+        elif not response["LayerVersions"]:
+            return None
+
+    def __create_new_lambda_layer(self, session):
```

Comment:
I suggest to depoly the rdklib lambda layer from severless application repository. You can create a changes set with https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/serverlessrepo.html#ServerlessApplicationRepository.Client.create_cloud_formation_change_set and then execute the change set with https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Client.execute_change_set. This will reduce the code and avoid creating a temporary S3 bucket